packages/headless/src/features/folding/folding-slice.ts

```diff
@@ -119,6 +120,13 @@ function createCollections(
     if (!collectionId) {
       return;
     }
+    if (result.parentResult) {
+      collections[collectionId] = createCollectionFromResult(
+        result,
+        fields,
+        result.parentResult
```

Comment:
It looks like this would break https://coveord.atlassian.net/browse/KIT-1569.

I think what you're looking for is to resolve the parent if it exists instead of the relevant result when the root cannot be resolved using the fields. If so, I think you should replace:
```ts
const resultToUseAsRoot =
  rootResult ??
  resolveRootFromFields(resultsInCollection, fields) ??
  relevantResult;
```
with:
```ts
const resultToUseAsRoot =
  rootResult ??
  resolveRootFromFields(resultsInCollection, fields) ??
  relevantResult.parentResult ??
  relevantResult;
```

Reply:
I should have written some tests for that!